opal/corelib/regexp.rb

```diff
@@ -119,6 +119,36 @@ def =~(string)
     match(string) && $~.begin(0)
   end
 
+  def freeze
+    # Specialized version of freeze, because the $$gm and $$g properties need to be set
+    # especially for RegExp. At the same time instance variables are ignored here.
+
+    return self if frozen?
+
+    %x{
+      self.$$frozen = true;
+
+      // set $$id
+      if (!self.hasOwnProperty('$$id')) { $prop(self, '$$id', $uid); }
+
+      // ensure $$meta can be set lazily, $$meta is frozen when set in runtime.js
+      if (!self.hasOwnProperty('$$meta')) { $prop(self, '$$meta', null); }
+
+      // $$comparable is used internally and set multiple times
+      // defining it before sealing ensures it can be modified later on
+      if (!self.hasOwnProperty('$$comparable')) { $prop(self, '$$comparable', null); }
+
```

Comment:
Perhaps consider using super in this function to make it smaller?

Reply:
Done